packages/core/js-client/src/util/testUtils.ts

```diff
@@ -146,69 +146,21 @@ class NoopConnection implements IConnection {
 }
 
 export class TestPeer extends FluencePeer {
-  constructor(keyPair: KeyPair, connection: IConnection) {
+  constructor(keyPair: KeyPair, connection: IConnection, marine: IMarineHost) {
     const jsHost = new JsServiceHost();
 
-    let marineJsWasm: ArrayBuffer;
-    let avmWasm: ArrayBuffer;
-
-    const marine = new MarineBackgroundRunner(
-      {
-        async getValue() {
-          // TODO: load worker in parallel with avm and marine, test that it works
-          return getWorker("@fluencelabs/marine-worker", "/");
-        },
-        start() {
-          return Promise.resolve(undefined);
-        },
-        stop() {
-          return Promise.resolve(undefined);
-        },
-      },
-      {
-        getValue() {
-          return marineJsWasm;
-        },
-        async start(): Promise<void> {
-          marineJsWasm = await fetchResource(
-            "@fluencelabs/marine-js",
-            "/dist/marine-js.wasm",
-            "/",
-          ).then((res) => {
-            return res.arrayBuffer();
-          });
-        },
-        stop(): Promise<void> {
-          return Promise.resolve(undefined);
-        },
-      },
-      {
-        getValue() {
-          return avmWasm;
-        },
-        async start(): Promise<void> {
-          avmWasm = await fetchResource(
-            "@fluencelabs/avm",
-            "/dist/avm.wasm",
-            "/",
-          ).then((res) => {
-            return res.arrayBuffer();
-          });
-        },
-        stop(): Promise<void> {
-          return Promise.resolve(undefined);
-        },
-      },
-    );
-
     super(DEFAULT_CONFIG, keyPair, marine, jsHost, connection);
   }
 }
 
 export const mkTestPeer = async () => {
   const kp = await KeyPair.randomEd25519();
   const conn = new NoopConnection();
-  return new TestPeer(kp, conn);
+
+  const marineDeps = await parallelLoadStrategy("/");
+  const marine = new MarineBackgroundRunner(...marineDeps);
```

Comment:
parallelLoadStrategy sound a bit generic and I don't know if it is a strategy. You actually load stuff with this function. maybe `loadMarineDependencies` I don't know. Also these two steps seem to be always together. Do you expect to ever need to call this function without passing args to MarineBackgroundRunner? In any case I would create a helper function e.g. `initMarine` or something that does both of these steps and returns marine

Reply:
Well, it's more about inversion of control. In other words, i'd like to make MarineBackgroundRunner unaware of how these deps are loaded. Thats why i want to explicitly do these on a separate lines. For example in unit tests these deps could be mocked or substituted somehow.

I cannot come up with more strategies for this case. So yeah, looks like it's better to avoid strategies and just use a helper function as you suggested.